### PR TITLE
Various sp

### PR DESCRIPTION
**Pull request by DantongWang**
Hi, finally I make the example codes readable. :)

### CODE REVIEW

File: CompCMD_LocationTM_OPTw.m [hunk 1]
```diff
@@ -0,0 +1,51 @@
+function CompCMD_LocationTM_OPTw(N, L)
+%% 
+% CompDMD_Location.m provides a method to calculate Sigma Points for a
```
Comment (yannikschaelte) [inline CompCMD_LocationTM_OPTw.m]:
```suggestion
% CompCMD_Location.m provides a method to calculate Sigma Points for a
```

File: CompCMD_LocationTM_OPTw.m [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+%   between dirac mixture approximation and normal distribution in
+%   terms of Localized Cumulative Distribution(LCD). Multistart method with
+%   20 strart points is used here.
+%
+% Parameters:
+%   N: dimension of dirac mixture approximation
+%   L: number of component for each dimension 
+%
+% Return values:
+%   optx: optimizition result of the location of dirac distribution
```
Comment (yannikschaelte) [inline CompCMD_LocationTM_OPTw.m]:
```suggestion
%   optx: optimization result of the location of dirac distribution
```

File: distanceDiracGaussianOPTw.m [hunk 3]
```diff
@@ -0,0 +1,296 @@
+function [D,G] = distanceDiracGaussianOPTw(xw,N,L)
+%% 
+% distanceDiracGaussian.m calculates the distance between dirac mixture
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m:3]: what does the OPTw in the title mean?

File: distanceDiracGaussianOPTw.m [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+    dD2dw = integral(@(b)dmd_dD2dwIntegral(N,x,b,w),0,3,'ArrayValue',true);
+end
+%calculate dD3dw
+dD3dw = pi^(N/2)*(sum(bsxfun(@times,w',(D31+D32)),1) + sum(bsxfun(@times,w,(D31+D32)),2)');
+Gw = -2*dD2dw + dD3dw;
+%% output
+D = D1-2*D2+D3;
+G = [reshape(G1+G2,[1,N*L]),Gw];
+end
+%%
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]:
```suggestion

```

File: distanceDiracGaussianOPTw.m [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+function D2Int = dmd_D2Integral(N,x,b,w)
+%% 
+% D2Integral.m calculates the value, which need to be integrated from 0 to
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]:
```suggestion
% D2Integral.m calculates the value, which needs to be integrated from 0 to
```

File: distanceDiracGaussianOPTw.m [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+% bmax(upper bound) to calculate D2, when dimention is odd.
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]:
```suggestion
% bmax(upper bound) to calculate D2, when dimension is odd.
```

File: distanceDiracGaussianOPTw.m [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+% History:
+% * 2018/01/05 Dantong Wang
+
+%% 
+sigma = zeros(N,1)+1;
+PI = prod(1./((sigma.^2+2*b.^2).^(1/2)));
+SIGMA = exp((-1/2)*sum(x.^2./(sigma.^2+2*b.^2),1));
+G1Int = b.^(N+1)./(sigma.^2+2*b.^2)*PI*SIGMA;
+end
+%%
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]:
```suggestion

```

File: distanceDiracGaussianOPTw.m [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+k = floor(N/2);
+l = N - k*2;
+% if N is even
+if l == 0
+    D2 = (2*pi)^(N/2)*sum(w.*(dmd_Jmm(k,bmax,c,divcb,ex_inb)-dmd_Jmm(k,0,c,divc0,ex_in0)),2);
+% if N is odd
+else
+    D2 = integral(@(b)dmd_D2Integral(N,x,b,w),0,3);
+end
+% calclate Tij as a matrix
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]:
```suggestion
% calculate Tij as a matrix
```

File: distanceDiracGaussianOPTw.m [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+% approximation and normal distribution in terms of Localized Cumulative
+% Distribution. Distance consists of three terms: D1,D2 and D3.
+% Detailed inference can be found in:
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% "Efficient Deterministic Dirac Mixture Approximation of Gaussian
+% Distributions", Igor Gilitschenski and Uwe D. Hanebeck
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%
+% Parameters:
+%   x: location of dirac distributions
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]: above, the parameter is called xw

File: distanceDiracGaussianOPTw.m [hunk 10]
```diff
@@ -0,0 +1,296 @@
...
+end
+%calculate dD3dw
+dD3dw = pi^(N/2)*(sum(bsxfun(@times,w',(D31+D32)),1) + sum(bsxfun(@times,w,(D31+D32)),2)');
+Gw = -2*dD2dw + dD3dw;
+%% output
+D = D1-2*D2+D3;
+G = [reshape(G1+G2,[1,N*L]),Gw];
+end
+%%
+function D2Int = dmd_D2Integral(N,x,b,w)
```
Comment (yannikschaelte) [inline distanceDiracGaussianOPTw.m]: in this file, there are a few functions starting with "dmd". should these be "cmd"?

File: distanceTM_OPTw.m [hunk 11]
```diff
@@ -0,0 +1,14 @@
+function [Dt,Gt] = distanceTM_OPTw(xw,N,L)
```
Comment (yannikschaelte) [inline distanceTM_OPTw.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/accuracyTestModel1.m [hunk 12]
```diff
@@ -0,0 +1,8 @@
+function [f,dfdphi] = accuracyTestModel1(phi)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/accuracyTestModel1.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/accuracyTestModel2.m [hunk 13]
```diff
@@ -0,0 +1,8 @@
+function [f,dfdphi] = accuracyTestModel2(phi)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/accuracyTestModel2.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/accuracyTestModel3.m [hunk 14]
```diff
@@ -0,0 +1,34 @@
+function [f,dfdphi] = accuracyTestModel3(phi)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/accuracyTestModel3.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/accuracyTest_main.m [hunk 15]
```diff
@@ -0,0 +1,205 @@
+function accuracyTest_main(Model)
+%% 
+% accuracyTest_main12.m computes the abosulute errors to the true mean and
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/accuracyTest_main.m]:
```suggestion
% accuracyTest_main.m computes the absolute errors to the true mean and 
```

File: examples/CMDaccuracyTest/average_difference.m [hunk 16]
```diff
@@ -0,0 +1,11 @@
+function [dApp_MCa] = average_difference(dApp_MC, dim)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/average_difference.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/average_time.m [hunk 17]
```diff
@@ -0,0 +1,4 @@
+function [my, Cy] = average_time(dApp_MCt)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/average_time.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/compMeanErrorbar.m [hunk 18]
```diff
@@ -0,0 +1,32 @@
+function [dApp_MC, dApp_MC_neg, dApp_MC_pos] = compMeanErrorbar(model, ...
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/compMeanErrorbar.m:1]: short explanation what this function does?

File: examples/CMDaccuracyTest/structminus.m [hunk 19]
```diff
@@ -0,0 +1,4 @@
+function dApp_MCP = structminus(App, MCP)
```
Comment (yannikschaelte) [inline examples/CMDaccuracyTest/structminus.m:1]: short explanation what this function does?

File: getSigmaPointApp.m [hunk 20]
```diff
@@ -332,9 +369,9 @@
 end
 
 SP.Y = Y;
-if(any(isnan(Y)))
-    error('Failed to successfully integrate system at all SigmaPoints')
-end
+% if(any(isnan(Y)))
```
Comment (yannikschaelte) [inline getSigmaPointApp.m]: why was this check deleted?

**Review by yannikschaelte — APPROVED** [review]
I don't know the project well enough to judge the content, so here are just some superficial remarks :)

File: distanceDiracGaussianOPTw.m (hunk 3 above)
Comment (DantongWang) [reply distanceDiracGaussianOPTw.m:3]: means to optimize weights as well...

**Pull request merged**